teerdays/src/tests.rs

```diff
@@ -0,0 +1,289 @@
+		assert_eq!(account_info.consumers, 1);
+		assert_eq!(account_info.data.frozen, alice_free);
+	})
+}
+
+#[test]
+fn bond_extra_works() {
+	new_test_ext().execute_with(|| {
+		run_to_block(1);
+		let now: Moment = 42;
+		set_timestamp(now);
+
+		let alice = AccountKeyring::Alice.to_account_id();
+		let amount: BalanceOf<Test> = 10_000_000_000_000;
+		assert_ok!(TeerDays::bond(RuntimeOrigin::signed(alice.clone()), amount));
+
+		run_to_block(2);
+		let now = now + 10_000;
+		set_timestamp(now);
+
+		let extra_amount = amount / 2;
+		assert_ok!(TeerDays::bond_extra(RuntimeOrigin::signed(alice.clone()), extra_amount));
+
+		let expected_event = RuntimeEvent::TeerDays(TeerDaysEvent::Bonded {
+			account: alice.clone(),
+			amount: extra_amount,
+		});
+		assert!(System::events().iter().any(|a| a.event == expected_event));
+
+		let teerdays = TeerDays::teerday_bonds(&alice)
+			.expect("TeerDays entry for bonded account should exist");
+		assert_eq!(teerdays.value, amount + extra_amount);
+		assert_eq!(teerdays.accumulated_tokentime, amount * 10_000);
+		assert_eq!(teerdays.last_updated, now);
+
+		let account_info = System::account(&alice);
+		assert_eq!(account_info.data.frozen, amount + extra_amount);
+	})
+}
+
+#[test]
+fn bond_extra_saturates_at_free_margin() {
+	new_test_ext().execute_with(|| {
+		run_to_block(1);
+		let now: Moment = 42;
+		set_timestamp(now);
+
+		let alice = AccountKeyring::Alice.to_account_id();
+		let alice_free: BalanceOf<Test> = 11_000_000_000_000;
+		<Test as pallet::Config>::Currency::make_free_balance_be(&alice, alice_free);
+		let amount: BalanceOf<Test> = 10_000_000_000_000;
+		assert_ok!(TeerDays::bond(RuntimeOrigin::signed(alice.clone()), amount));
+
+		let teerdays = TeerDays::teerday_bonds(&alice)
+			.expect("TeerDays entry for bonded account should exist");
+		assert_eq!(teerdays.value, amount);
+		assert_eq!(teerdays.accumulated_tokentime, 0);
+		assert_eq!(teerdays.last_updated, now);
+
+		let account_info = System::account(&alice);
+		assert_eq!(account_info.consumers, 1);
+		assert_eq!(account_info.data.frozen, amount);
+
+		run_to_block(2);
+		let now = now + 10_000;
+		set_timestamp(now);
+
+		let extra_amount = amount / 2;
+		assert_ok!(TeerDays::bond_extra(RuntimeOrigin::signed(alice.clone()), extra_amount));
+
+		let expected_event = RuntimeEvent::TeerDays(TeerDaysEvent::Bonded {
+			account: alice.clone(),
+			amount: 1_000_000_000_000,
+		});
+		assert_eq!(System::events().get(1).unwrap().event, expected_event);
+
+		let teerdays = TeerDays::teerday_bonds(&alice)
+			.expect("TeerDays entry for bonded account should exist");
+		assert_eq!(teerdays.value, amount + 1_000_000_000_000);
+		assert_eq!(teerdays.accumulated_tokentime, amount * 10_000);
+		assert_eq!(teerdays.last_updated, now);
+
+		let account_info = System::account(&alice);
+		assert_eq!(account_info.data.frozen, amount + 1_000_000_000_000);
+	})
+}
+
+#[test]
+fn unbonding_and_delayed_withdraw_works() {
```

Comment:
For me would be more concise to say:

`withrawing_unbonded_after_unlock_period_works`